components/form.js

```diff
@@ -0,0 +1,143 @@
+import React from "react";
+import Link from "next/link";
+
+export default class Form extends React.Component {
+  constructor() {
+    super();
+    this.state = { token: "", formFields: [], formData: {} };
+  }
+
+  // Make state targets equal to the value of input fields
+  change = e => {
+    this.setState({
+      formData: {
+        ...this.state.formData,
+        [e.target.name]: e.target.value
+      }
+    });
+  };
+
+  componentDidMount() {
+    // Fetch the data form fields from API, use it in state
+    let formFields = JSON.parse(localStorage.getItem("Form Fields"));
+    if (!(Array.isArray(formFields) && formFields.length)) {
```

Comment:
ah , this is really nice because it leverages localStorage to get things. The problem I see here is that it doesn't account for the form id. So form 15 and form 14 both have different fields. That should be accounted for when retrieving from local storage. 
ALSO, I'd set a timestamp to invalidate this cache in like 24hs or something (in a later iteration I'd depend on cache expiration headers from the  server)